portal/src/error/apps.ts

```diff
@@ -2,3 +2,13 @@ export interface APIDuplicatedAppIDError {
   errorName: "AlreadyExists";
   reason: "DuplicatedAppID";
 }
+
+export interface APIReservedAppIDError {
+  errorName: "Forbidden";
+  reason: "AppIDReserved";
+}
+
+export interface APIInvalidAppIDError {
+  errorName: "Invalid";
```

Comment:
The invalid app ID error is currently a generic error (i.e. `apierrors.NewInvalid`) produced by `AppService`. Since we need to interpret the error programmatically now, we should assign a correct reason (e.g. `InvalidAppID`) at server side like the reserved app ID error above.